omc3/tune_analysis/kick_file_modifiers.py

```diff
@@ -0,0 +1,248 @@
+    plus the standard deviation of the moving average.
+
+    Args:
+        kickac_df: Dataframe containing the data
+
+    Returns:
+        Modified kick_ac
+    """
+    for plane in PLANES:
+        kickac_df[get_corr_natq_err_col(plane)] = np.sqrt(
+            np.power(kickac_df[get_natq_err_col(plane)], 2) +
+            np.power(kickac_df[get_mav_err_col(plane)], 2)
+        )
+    return kickac_df
+
+
+# Data Extraction ##############################################################
+
+
+def get_odr_data(kickac_df, action_plane, tune_plane, order, corrected=False):
+    """ Extract the data from kickac.
+
+    Args:
+        kickac_df: Dataframe containing the data
+        action_plane: Plane of the action
+        tune_plane: Plane of the tune
+        order: Order of the odr fit
+        corrected: (BBQ) corrected data or uncorrected fit?
+
+    Returns:
+        Dictionary containing fit data from odr
+
+    """
+
+    header_val, header_err = _get_odr_headers(corrected)
+    odr_data = DotDict(beta=[0] * (order+1), sd_beta=[0] * (order+1))
+    for idx in range(order+1):
+        odr_data.beta[idx] = kickac_df.headers[header_val(action_plane, tune_plane, idx)]
+        odr_data.sd_beta[idx] = kickac_df.headers[header_err(action_plane, tune_plane, idx)]
+    return odr_data
+
+
+def get_ampdet_data(kickac_df, action_plane, tune_plane, corrected=False):
+    """ Extract the data needed for the (un)corrected amplitude detuning
+    from the kickac dataframe.
+
+    Args:
+        kickac_df: Dataframe containing the data
+        action_plane: Plane of the action
+        tune_plane: Plane of the tune
+
+
+    Returns:
+        Dictionary containing action, tune, action_err and tune_err
+
+    """
+    col_natq, col_natq_std = _get_ampdet_columns(corrected)
+
+    columns = {"action": get_action_col(action_plane),
+               "action_err": get_action_err_col(action_plane),
+               "tune": col_natq(tune_plane),
+               "tune_err": col_natq_std(tune_plane),
+               }
+
+    not_found = [cv for cv in columns.values() if cv not in kickac_df.columns]
+    if any(not_found):
+        raise KeyError(f"The following columns were not found in kick-file: '{str(not_found)}'")
+
+    data = kickac_df.loc[:, list(columns.values())]
+    data.columns = columns.keys()
+
+    if data.isna().any().any():
+        LOG.warn(
+            f"Amplitude Detuning data for Q{tune_plane} and J{action_plane} contains NaNs"
+        )
+        data = data.dropna(axis=0)
+    return data.to_dict('series')
+
+
+# Timed DataFrames -------------------------------------------------------------
+
+
+def get_timestamp_index(index):
+    return pd.Index([i.timestamp() for i in index])
+
+
+def read_timed_dataframe(path):
+    df = tfs.read(path, index=get_time_col())
+    df.index = pd.Index([CERNDatetime.from_cern_utc_string(i) for i in df.index], dtype=object)
+    return df
+
+
+def write_timed_dataframe(path, df):
+    df = df.copy()
```

Comment:
This might be changing behaviour with pandas version around 1.0

Reply:
to be checked then